React/Views/RCTModalHostViewManager.m

```diff
@@ -89,7 +89,7 @@ - (void)dismissModalHostView:(RCTModalHostView *)modalHostView withViewControlle
   if (_dismissalBlock) {
     _dismissalBlock([modalHostView reactViewController], viewController, animated, completionBlock);
   } else {
-    [viewController dismissViewControllerAnimated:animated completion:completionBlock];
+    [viewController.presentingViewController dismissViewControllerAnimated:animated completion:completionBlock];
```

Comment:
The documentation is clear and confirms my recollections:
```
dismissViewControllerAnimated:completion:

Dismisses the view controller that was presented modally by the view controller.
SDKs
```

So, the fix looks perfectly legit to me, the only Q that prevents me from landing it:
I don't understand how/why it worked previously. Did it? (Until we find answer to this question we cannot be sure that our conclusions are correct.)

Reply:
@shergin Could be why it works, but maybe it dismisses the wrong vc sometimes.

```
 If you call this method on the presented view controller itself, UIKit asks the presenting view controller to handle the dismissal.
```

Thanks for clarifiying what `reactViewController` is. LGTM

Reply:
If we just have one Modal, it works whatever we passed presented VC or presenting VC, but if we have multiple Modals, bug may happen, for example, VC present Modal A, ModalA present Modal B, if we call [ModalA dismissVC], what this should perform? because `ModalA` is the presenting VC to present `ModalB`,but it also the presented VC which presenting VC is the closest VC when present ModalA, I think this is why bug happened. :)